src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs

```diff
@@ -360,5 +360,228 @@ public OpenIdConnectConfiguration(string json)
         /// </summary>
         [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore, NullValueHandling = NullValueHandling.Ignore, PropertyName = OpenIdProviderMetadataNames.UserInfoSigningAlgValuesSupported, Required = Required.Default)]
         public ICollection<string> UserInfoEndpointSigningAlgValuesSupported { get; } = new Collection<string>();
+
+#region shouldserialize
+        /// <summary>
+        /// Gets a bool that determines if the 'acr_values_supported' (AcrValuesSupported) property should be serialized.
+        /// This is used by Json.NET in order to conditionally serialize properties.
+        /// </summary>
+        /// <return>true if 'acr_values_supported' (AcrValuesSupported) is not empty; otherwise, false.</return>
+        public bool ShouldSerializeAcrValuesSupported()
```

Comment:
Do you have to make these methods public? I would suggest you try making them non-public (private/internal) and see if JSON.NET respects that. That way you avoid polluting the public API with all these definitions.

Reply:
Ignore my comment about this. Sadly they have to be public

Reply:
Alternative idea, I would suggest putting `[EditorBrowsable(EditorBrowsableState.Never)]` so that at least they don't show up on intellisense.

Reply:
The main reason is that users are not intended to call these methods.

Reply:
:-( yes sadly they have to be public. 
+1 on the [EditorBrowsable(EditorBrowsableState.Never)]